{{cookiecutter.project_shortname}}/cookiecutter_templates/AsyncFunctionComponent.react.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { {{cookiecutter.component_name}} as RealComponent } from '../LazyLoader';
+
+/**
+ * ExampleComponent is an example component.
+ * It takes a property, `label`, and
+ * displays it.
+ * It renders an input with the property `value`
+ * which is editable by the user.
+ */
+const {{cookiecutter.component_name}} = (props) => {
+    return (
+        <React.Suspense fallback={null}>
+            <RealComponent {...props}/>
+        </React.Suspense>
+    );
+};
```

Comment:
There isn't really any difference in both AsyncClassComponent and AsyncFunctionComponent, they are also unlikely to change by the user as the real code reside in the fragments directory. We can keep only one of the async and have the `component_type` only for the `Class` or `Function` component.

Reply:
sounds good! I'm leaning towards keeping the `AsyncFunctionComponent` since functional components are now the official type in react documentation.